contracts/croncat-agents/src/contract.rs

```diff
@@ -1,31 +1,538 @@
+use std::cmp;
+use std::ops::Div;
+
+use crate::balancer::Balancer;
+use crate::error::ContractError;
+use crate::msg::{
+    AgentContractInstantiateMsg as InstantiateMsg, 
+    AgentExecuteMsg as ExecuteMsg,
+    AgentQueryMsg as QueryMsg,
+};
+use cw2::set_contract_version;
+
+use crate::state::{
+    ACTIVE_AGENTS, AGENTS, AGENT_BALANCER, AGENT_NOMINATION_BEGIN_TIME, CONFIG, PENDING_AGENTS,
+};
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cosmwasm_std::{
+    has_coins, to_binary, Addr, BankMsg, Binary, Coin, Deps, DepsMut, Env, MessageInfo, Response,
+    StdError, StdResult, Storage, SubMsg, Uint128,
+};
+use croncat_sdk_agents::msg::{AgentResponse, AgentTaskResponse, GetAgentIdsResponse};
+use croncat_sdk_agents::types::{Agent, AgentStatus};
+use croncat_sdk_core::types::Config;
```

Comment:
```suggestion
use croncat_sdk_agents::types::Config;
```
Agents should have it's own config!

Reply:
yes, but from this config I get native_denom

Reply:
> yes, but from this config I get native_denom

Why do you need native_denom?

Reply:
for sending rewards